docs/snippets/web-components/button-story-component-decorator.ts.mdx

```diff
@@ -0,0 +1,29 @@
+```ts
+// demo-button.stories.ts
+import type { Meta, StoryObj } from '@storybook/web-components';
+
+import { html } from 'lit-html';
+
+import './demo-button';
+
+const meta: Meta = {
+  /* 👇 The title prop is optional.
+   * See https://storybook.js.org/docs/7.0/web-components/configure/overview#configure-story-loading
+   * to learn how to generate automatic titles
+   */
+  title: 'Button',
+  decorators: [(story) => html`<div style="margin: 3em">${story()}</div>`],
+};
+export default meta;
+
+type Story = StoryObj;
```

Comment:
When the component doesn't have a prop type available (e.g. `import './demo-button'` vs. `import { Button, ButtonProps } from './demo-button'`) then there's nothing to pass to `StoryObj` and this assignment is a bit silly. But it's also more consistent across situations. Wdyt?

Reply:
I know, and I had the same question as you did, and I went for consistency and left in the StoryObj type as an additional precaution to prevent unforeseen questions about missing typings.

Reply:
To be clear, the alternatives are:

A:
```js
type Story = StoryObj;

export const Example: Story = { ... }
```

or

B:
```js
export const Example: StoryObj = { ... }
```

A is more consistent with the more prevalent case of `type Story = StoryObj<ComponentProps>`, but without passing a generic, the assignment could be seen as a bit silly or unnecessary. Hence, B.

Looking at your PR, it looks like you might've gone with B?

Reply:
To some degree, yes I went with that, but it's a bit situational and something we'll need to consider moving forward as we build more accurate examples